### ISSUE TITLE
Consider not relying on constrained layout

### ISSUE THREAD

**Issue by jklymak** [matplotlib/grid-strategy#54]
Overall this looks fun.

Super happy you are using constrained layout, but suggest you make it optional somehow. There are some times when it'll fail and users will want a way to turn it off. 

I'm also not sure about initializing the figure inside of your helper, instead of letting the user pass it (optionally?).

**Comment by pganssle** [comment]
Agreed, honestly I kinda thought we already *were* taking the figure as input.

This issue is related to #47, since I'm not sure if all plotting libraries have the concept of a figure that can be passed in.
